### ISSUE TITLE
msi installer error versions 0.13.0 and 0.14.0 - service fails to start

### ISSUE THREAD

**Issue by lukeg87** [prometheus-community/windows_exporter#627]
I'm having this problem with one Server 2016 CU 2020-08 in our environment, versions 0.13.0 and 0.14.0

`Product: windows_exporter -- Error 1920. Service 'windows_exporter' (windows_exporter) failed to start.  Verify that you have sufficient privileges to start system services.`

Tried installing using my domain admin account, local administrator and as a scheduled task as SYSTEM.

The installer gets stuck at starting the service which eventually fails. Older versions install but we've migrated the entire estate to windows_exporter now :/

luckily this is just a sandbox server. I can't find any other logs apart from msi installer errors

```
=== Logging started: 10/14/2020  11:48:06 ===
Action start 11:48:06: INSTALL.
Action start 11:48:06: FindRelatedProducts.
Action ended 11:48:06: FindRelatedProducts. Return value 1.
Action start 11:48:06: LaunchConditions.
Action ended 11:48:06: LaunchConditions. Return value 1.
Action start 11:48:06: ValidateProductID.
Action ended 11:48:06: ValidateProductID. Return value 1.
Action start 11:48:06: CostInitialize.
Action ended 11:48:06: CostInitialize. Return value 1.
Action start 11:48:06: FileCost.
Action ended 11:48:06: FileCost. Return value 1.
Action start 11:48:06: CostFinalize.
Action ended 11:48:06: CostFinalize. Return value 1.
Action start 11:48:06: MigrateFeatureStates.
Action ended 11:48:06: MigrateFeatureStates. Return value 0.
Action start 11:48:06: InstallValidate.
Action ended 11:48:06: InstallValidate. Return value 1.
Action start 11:48:06: InstallInitialize.
Action ended 11:48:06: InstallInitialize. Return value 1.
Action start 11:48:06: ProcessComponents.
Action ended 11:48:07: ProcessComponents. Return value 1.
Action start 11:48:07: UnpublishFeatures.
Action ended 11:48:07: UnpublishFeatures. Return value 1.
Action start 11:48:07: StopServices.
Action ended 11:48:07: StopServices. Return value 1.
Action start 11:48:07: DeleteServices.
Action ended 11:48:07: DeleteServices. Return value 1.
Action start 11:48:07: RemoveRegistryValues.
Action ended 11:48:07: RemoveRegistryValues. Return value 1.
Action start 11:48:07: WixSchedFirewallExceptionsUninstall.
SchedFirewallExceptions:  Component 'windows_exporter.exe' action state (1) doesn't match request (2)
SchedFirewallExceptions:  No firewall exceptions scheduled
Action ended 11:48:07: WixSchedFirewallExceptionsUninstall. Return value 1.
Action start 11:48:07: RemoveFiles.
Action ended 11:48:07: RemoveFiles. Return value 0.
Action start 11:48:07: RemoveFolders.
Action ended 11:48:07: RemoveFolders. Return value 1.
Action start 11:48:07: CreateFolders.
Action ended 11:48:07: CreateFolders. Return value 1.
Action start 11:48:07: InstallFiles.
Action ended 11:48:07: InstallFiles. Return value 1.
Action start 11:48:07: SetCollectorsFlag.
Action ended 11:48:07: SetCollectorsFlag. Return value 1.
Action start 11:48:07: SetListenFlag.
Action ended 11:48:07: SetListenFlag. Return value 1.
Action start 11:48:07: WixSchedFirewallExceptionsInstall.
SchedFirewallExceptions:  Scheduling firewall exception (1€windows_exporter (HTTP 9182)€2147483647€*€1€2€c:\Program Files\windows_exporter\windows_exporter.exe€9182€6€windows_exporter HTTP endpoint)
Action start 11:48:07: WixRollbackFirewallExceptionsInstall.
Action ended 11:48:07: WixRollbackFirewallExceptionsInstall. Return value 1.
Action start 11:48:07: WixExecFirewallExceptionsInstall.
Action ended 11:48:07: WixExecFirewallExceptionsInstall. Return value 1.
Action ended 11:48:07: WixSchedFirewallExceptionsInstall. Return value 1.
Action start 11:48:07: WriteRegistryValues.
Action ended 11:48:07: WriteRegistryValues. Return value 1.
Action start 11:48:07: InstallServices.
Action ended 11:48:07: InstallServices. Return value 1.
Action start 11:48:07: SchedServiceConfig.
Action start 11:48:07: RollbackServiceConfig.
Action ended 11:48:07: RollbackServiceConfig. Return value 1.
Action start 11:48:07: ExecServiceConfig.
Action ended 11:48:07: ExecServiceConfig. Return value 1.
Action ended 11:48:07: SchedServiceConfig. Return value 1.
Action start 11:48:07: StartServices.
Action ended 11:48:07: StartServices. Return value 1.
Action start 11:48:07: RegisterUser.
Action ended 11:48:07: RegisterUser. Return value 1.
Action start 11:48:07: RegisterProduct.
Action ended 11:48:07: RegisterProduct. Return value 1.
Action start 11:48:07: PublishFeatures.
Action ended 11:48:07: PublishFeatures. Return value 1.
Action start 11:48:07: PublishProduct.
Action ended 11:48:07: PublishProduct. Return value 1.
Action start 11:48:07: InstallExecute.
ExecFirewallExceptions:  Installing firewall exception2 windows_exporter (HTTP 9182) (c:\Program Files\windows_exporter\windows_exporter.exe)
MSI (s) (B0:00) [11:54:31:879]: Product: windows_exporter -- Error 1920. Service 'windows_exporter' (windows_exporter) failed to start.  Verify that you have sufficient privileges to start system services.
 
Error 1920. Service 'windows_exporter' (windows_exporter) failed to start.  Verify that you have sufficient privileges to start system services.
Are you sure you want to cancel?
Action ended 11:54:32: InstallExecute. Return value 3.
ExecFirewallExceptions:  Uninstalling firewall exception2 windows_exporter (HTTP 9182) (c:\Program Files\windows_exporter\windows_exporter.exe)
Action ended 11:54:33: INSTALL. Return value 3.
Property(S): UpgradeCode = {66A6EB5B-1FC2-4B14-A362-5CEEC6413308}
Property(S): WixRollbackFirewallExceptionsInstall = 1€windows_exporter (HTTP 9182)€2147483647€*€1€2€c:\Program Files\windows_exporter\windows_exporter.exe€9182€6€windows_exporter HTTP endpoint
Property(S): WixExecFirewallExceptionsInstall = 1€windows_exporter (HTTP 9182)€2147483647€*€1€2€c:\Program Files\windows_exporter\windows_exporter.exe€9182€6€windows_exporter HTTP endpoint
Property(S): APPLICATIONROOTDIRECTORY = c:\Program Files\windows_exporter\
Property(S): textfile_inputs = c:\Program Files\windows_exporter\textfile_inputs\
Property(S): CollectorsFlag = --collectors.enabled cpu,cs,logical_disk,net,os,service,memory,system,mssql,process
Property(S): ListenFlag = --telemetry.addr :9182
Property(S): ExecServiceConfig = SchedServiceConfig€windows_exporter€1€restart€restart€restart€0€5€€
Property(S): RollbackServiceConfig = SchedServiceConfig
Property(S): ProgramFiles64Folder = c:\Program Files\
Property(S): TARGETDIR = D:\
Property(S): SourceDir = c:\users\[redacted]\downloads\
Property(S): VersionNT = 603
Property(S): ENABLED_COLLECTORS = cpu,cs,logical_disk,net,os,service,memory,system,mssql,process
Property(S): ALLUSERS = 1
Property(S): LISTEN_PORT = 9182
Property(S): Manufacturer = prometheus-community
Property(S): ProductCode = {287A67BF-0D4D-4DB2-AB1D-97EB8A098C29}
Property(S): ProductLanguage = 1033
Property(S): ProductName = windows_exporter
Property(S): ProductVersion = 0.13.0
Property(S): SecureCustomProperties = ENABLED_COLLECTORS;EXTRA_FLAGS;LISTEN_ADDR;LISTEN_PORT;METRICS_PATH;TEXTFILE_DIR;WIX_DOWNGRADE_DETECTED;WIX_UPGRADE_DETECTED
Property(S): MsiLogFileLocation = c:\windows-exporter-log.txt
Property(S): PackageCode = {FE721A7B-2F5E-41B3-936B-9F5109D36AB7}
Property(S): ProductState = -1
Property(S): PackagecodeChanging = 1
Property(S): REBOOT = ReallySuppress
Property(S): CURRENTDIRECTORY = C:\Windows\system32
Property(S): CLIENTUILEVEL = 2
Property(S): CLIENTPROCESSID = 3604
Property(S): MsiSystemRebootPending = 1
Property(S): VersionDatabase = 200
Property(S): VersionMsi = 5.00
Property(S): VersionNT64 = 603
Property(S): WindowsBuild = 9600
Property(S): ServicePackLevel = 0
Property(S): ServicePackLevelMinor = 0
Property(S): MsiNTProductType = 3
Property(S): WindowsFolder = c:\Windows\
Property(S): WindowsVolume = c:\
Property(S): System64Folder = C:\Windows\system32\
Property(S): SystemFolder = c:\Windows\SysWOW64\
Property(S): RemoteAdminTS = 1
Property(S): ProgramFilesFolder = C:\Program Files (x86)\
Property(S): CommonFilesFolder = C:\Program Files (x86)\Common Files\
Property(S): CommonFiles64Folder = C:\Program Files\Common Files\
Property(S): TemplateFolder = C:\ProgramData\Microsoft\Windows\Templates\
Property(S): CommonAppDataFolder = C:\ProgramData\
Property(S): AdminToolsFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Administrative Tools\
Property(S): StartupFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Startup\
Property(S): ProgramMenuFolder = C:\ProgramData\Microsoft\Windows\Start Menu\Programs\
Property(S): StartMenuFolder = C:\ProgramData\Microsoft\Windows\Start Menu\
Property(S): DesktopFolder = C:\Users\Public\Desktop\
Property(S): FontsFolder = C:\Windows\Fonts\
Property(S): GPTSupport = 1
Property(S): OLEAdvtSupport = 1
Property(S): ShellAdvtSupport = 1
Property(S): MsiAMD64 = 6
Property(S): Msix64 = 6
Property(S): Intel = 6
Property(S): PhysicalMemory = 32768
Property(S): VirtualMemory = 35290
Property(S): AdminUser = 1
Property(S): MsiTrueAdminUser = 1
Property(S): UserLanguageID = 1033
Property(S): ComputerName = ART-MSSQL-SB
Property(S): SystemLanguageID = 1033
Property(S): ScreenX = 1024
Property(S): ScreenY = 768
Property(S): CaptionHeight = 23
Property(S): BorderTop = 1
Property(S): BorderSide = 1
Property(S): MsiTabletPC = 1
Property(S): TextHeight = 16
Property(S): TextInternalLeading = 3
Property(S): ColorBits = 32
Property(S): TTCSupport = 1
Property(S): Time = 11:54:33
Property(S): Date = 10/14/2020
Property(S): MsiNetAssemblySupport = 4.7.3062.0
Property(S): MsiWin32AssemblySupport = 6.3.14393.0
Property(S): RedirectedDllSupport = 2
Property(S): MsiRunningElevated = 1
Property(S): Privileged = 1
Property(S): USERNAME = Windows User
Property(S): DATABASE = c:\Windows\Installer\d0598.msi
Property(S): UILevel = 3
Property(S): ACTION = INSTALL
Property(S): ROOTDRIVE = D:\
Property(S): CostingComplete = 1
Property(S): OutOfDiskSpace = 0
Property(S): OutOfNoRbDiskSpace = 0
Property(S): PrimaryVolumeSpaceAvailable = 0
Property(S): PrimaryVolumeSpaceRequired = 0
Property(S): PrimaryVolumeSpaceRemaining = 0
Property(S): INSTALLLEVEL = 1
Property(S): SourcedirProduct = {287A67BF-0D4D-4DB2-AB1D-97EB8A098C29}
Property(S): ProductToBeRegistered = 1
MSI (s) (B0:00) [11:54:33:098]: Product: windows_exporter -- Installation failed.
 
MSI (s) (B0:00) [11:54:33:098]: Windows Installer installed the product. Product Name: windows_exporter. Product Version: 0.13.0. Product Language: 1033. Manufacturer: prometheus-community. Installation success or error status: 1603.
 
=== Logging stopped: 10/14/2020  11:54:33 ===


```

**Comment by carlpett** [comment]
Hi @lukeg87,
Is this an upgrade or a fresh install?

**Comment by lukeg87** [comment]
Hi @carlpett, 

The server previously had wmi_exporter v 0.7.0 installed. During deployment, i used the msi product codes to remove the old wmi_exporters before installing v0.14.0

**Comment by ashishksinghin** [comment]
Do we have solution for this issue? facing same issue

**Comment by allthegoodnamesaretaken** [comment]
Facing the same issue with 0.18.1

**Comment by TCC421** [comment]
I am having the same issue. I believe I had version 0.13.0 installed before trying to install 0.20.0

**Comment by Sokan2402** [comment]
I had the same problem. Managed to fix it.
This error seems to be very generic so I have started checking few things:
•	Checking privileges via the Group policy configuration as per this tutorial:	https://stackoverflow.com/questions/20061057/error-1920-service-failed-to-start-verify-that-you-have-sufficient-privileges-t
•	Checked if nothing else is using the default port 9182 with: “netstat -aon”
•	Trying installing via command line with different parameters: “msiexec /i <path-to-msi-file> ENABLED_COLLECTORS=os,iis LISTEN_PORT=5000”
•	I have disabled the Firewall.
•	I have disabled the Antivirus real time protection.
•	Tried to install versions windows_exporter 0.18 to 0.21

I was still getting the error

I have decided to investigate the service before clicking “Cancel” as that would finish the installation and remove the windows_exporter service.
Checking the windows_exporter service properties revealed that it has a dependency - WMI Performance Adapter service.
The WMI Performance Adapter service was not present in my services.
I have downloaded and installed this service following this guide (for windows 10 Pro):
http://batcmd.com/windows/10/services/wmiapsrv/
Downloaded the file RestoreWMIPerformanceAdapterWindows10.bat
I have ran it with PowerShell
 ./RestoreWMIPerformanceAdapterWindows10.bat

After restart I have manually started the WMI Performance Adapter service and changed to automatic start.
The windows_exporter installed now successfully.
